blazar/plugins/floatingips/floatingip_plugin.py

```diff
@@ -75,6 +76,90 @@ def check_params(self, values):
             if not (netutils.is_valid_ipv4(ip) or netutils.is_valid_ipv6(ip)):
                 raise manager_ex.InvalidIPFormat(ip=ip)
 
+    def _update_allocations(self, dates_before, dates_after, reservation_id,
+                            reservation_status, fip_reservation, values,
+                            lease):
+        allocs = db_api.fip_allocation_get_all_by_values(
+            reservation_id=reservation_id)
+
+        allocs_to_remove = self._allocations_to_remove(
+            dates_before, dates_after, allocs)
+
+        if (allocs_to_remove and
+                reservation_status == status.reservation.ACTIVE):
+            raise manager_ex.NotEnoughFloatingIPAvailable()
+
+        kept_fips = len(allocs) - len(allocs_to_remove)
+        fip_ids_to_add = []
+        if kept_fips < 1:
+            min_fips = 1 - kept_fips if (1 - kept_fips) > 0 else 0
+            required_fips = values.get('required_fips', [])
+            amount = int(values['amount'])
+            fip_ids_to_add = self._matching_fips(
+                values['floatingip_id'], required_fips, amount,
+                dates_after['start_date'], dates_after['end_date'])
+
+            if len(fip_ids_to_add) < min_fips:
+                raise manager_ex.NotEnoughFloatingIPAvailable()
+
+        allocs_to_keep = [a for a in allocs if a not in allocs_to_remove]
+        allocs_to_add = [{'floatingip_id': f} for f in fip_ids_to_add]
+        new_allocations = allocs_to_keep + allocs_to_add
+
+        try:
+            self.usage_enforcer.check_usage_against_allocation_post_update(
+                values, lease, allocs, new_allocations)
+        except manager_ex.RedisConnectionError:
+            pass
+
+        for fip_id in fip_ids_to_add:
+            LOG.debug('Adding floating ip {} to reservation {}'.format(
+                fip_id, reservation_id))
+
+        for allocation in allocs_to_remove:
+            LOG.debug('Removing floating ip {} from reservation {}'.format(
+                allocation['floatingip_id'], reservation_id))
+
+    def _allocations_to_remove(self, dates_before, dates_after, allocs):
+        """Find candidate floating ip allocations to remove."""
+        allocs_to_remove = []
+        requested_fip_ids = [fip['id'] for fip in db_api.floatingip_list()]
+
+        for alloc in allocs:
+            if alloc['floatingip_id'] not in requested_fip_ids:
```

Comment:
Yea, this may be unnecessary for floating ips. It occurs after both oshosts and networks filter available resources by the parameters in in resource properties, which floating ips does not have.